manifests/init.pp

```diff
@@ -66,12 +81,13 @@
 # @param package_ensure
 #   The 'ensure' value for package resources
 class simp_firewalld (
+  Optional[Hash]                                       $rules,               # data in module
```

Comment:
This should be `Hash`, not `Optional[Hash]`, since it is set to `{}` by default in hiera.  (If it is `undef`, the code will fail on [line 158](https://github.com/simp/pupmod-simp-simp_firewalld/pull/39/files#diff-c6c3a824d185ff01e063011517a1b13a604dc3d7b63d1c67a30451c10df02bcbR158).)